### PR TITLE
style: adding annotations @server/@client to seperate client-specific and server-specific code (at_persistence_secondary)

### LINKED ISSUE #725: Clarify client-specific vs server-specific code in at_persistence_secondary_server
@VJag wrote:
> I agree. To start with should we introduce annotations like @client @server and annotate the code?

_Originally posted by @VJag in https://github.com/atsign-foundation/at_server/issues/723#issuecomment-1142264402_

### PR DESCRIPTION
**Pull request by srieteja**
closes https://github.com/atsign-foundation/at_server/issues/725
**- What I did**
added annotations for at_persistence_secondary methods
**- How I did it**
created custom annotation classes in at_persistence_spec
referred these classes as annotations to public methods using static consts
**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_secondary/at_persistence_secondary_server/lib/src/config/at_config.dart [hunk 1]
```diff
@@ -14,6 +12,7 @@ class AtConfig {
 
   ///stores 'Configuration' type under [configkey] in secondary.
   String configKey = 'configKey';
+  static const server = AtServerAnnotation();
```
Comment (gkc) [inline at_secondary/at_persistence_secondary_server/lib/src/config/at_config.dart]: The line of code above is being added in many files. These `server` and `client` top-level constants should instead be exported by the package - see https://pub.dev/documentation/meta/latest/meta/visibleForTesting-constant.html for example in the dart `meta` package

**Review by gkc — CHANGES_REQUESTED** [review]

File: at_secondary/at_persistence_secondary_server/lib/src/compaction/at_compaction_stats_service.dart [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import '../../at_persistence_secondary_server.dart';
```
Comment (murali-shris) [inline at_secondary/at_persistence_secondary_server/lib/src/compaction/at_compaction_stats_service.dart]: do we need this import ?

**Review by murali-shris — CHANGES_REQUESTED** [review]

File: at_secondary/at_persistence_secondary_server/lib/src/compaction/at_compaction_stats_service.dart (hunk 2 above)
Comment (srieteja) [reply at_secondary/at_persistence_secondary_server/lib/src/compaction/at_compaction_stats_service.dart]: No we do not. I have removed that in the latest commit.

File: at_secondary/at_persistence_secondary_server/lib/src/config/at_config.dart (hunk 1 above)
Comment (srieteja) [reply at_secondary/at_persistence_secondary_server/lib/src/config/at_config.dart]: I removed these initialisations and it looks like a couple of them were left out. In the recent commits, I have removed all such initializations.

**Review by sitaram-kalluri — APPROVED** [review]

**Review by gkc — APPROVED** [review]

**Review by murali-shris — APPROVED** [review]

**Pull request merged**